File: extract_pdf.js
const fs = require('fs');
const path = require('path');
const PDFParser = require('pdf2json');

const filePath = path.join(process.cwd(), 'real_test.pdf');
const parser = new PDFParser(null, 1);

parser.on('pdfParser_dataError', (errData) => {
  console.error('PDF Parse Error:', errData.parserError);
});

parser.on('pdfParser_dataReady', (pdfData) => {
  try {
    let text = '';
    const pages = pdfData.Pages;
    
    if (pages && pages.length > 0) {
      pages.forEach((page) => {
        if (page.Texts) {
          page.Texts.forEach((textObj) => {
            if (textObj.R) {
              textObj.R.forEach((run) => {
                if (run.T) {
                  text += decodeURIComponent(run.T) + ' ';
                }
              });
            }
          });
          text += '\n';
        }
      });
    }
    
    console.log(text || 'No text extracted');
  } catch (err) {
    console.error('Error parsing PDF data:', err.message);
  }
});

fs.readFile(filePath, (err, data) => {
  if (err) {
    console.error('Error reading file:', err);
    return;
  }
  parser.parseBuffer(data);
});

File: extract_pdf.py
import pypdf
import sys

try:
    pdf_path = "real_test.pdf"
    with open(pdf_path, "rb") as file:
        pdf_reader = pypdf.PdfReader(file)
        text = ""
        for page_num in range(len(pdf_reader.pages)):
            page = pdf_reader.pages[page_num]
            text += page.extract_text()
        print(text)
except Exception as e:
    print(f"Error: {e}", file=sys.stderr)
